crawl-ref/source/timed-effects.cc

```diff
@@ -1417,28 +1414,31 @@ void update_level(int elapsedTime)
     delete_all_clouds();
 }
 
+/**
+ * Update the monster upon the player's return
+ *
+ * @param mon   The monster to update.
+ * @param turns How many turns the monster has been away from the player.
+ * @returns     Returns nullptr if monster is not updated.
+ *              Returns the updated monster if it was updated.
+ */
 monster* update_monster(monster& mon, int turns)
 {
-    mprf(MSGCH_WARN, "placing %s, %d", mon.name(DESC_PLAIN).c_str(),turns);
     // Pacified monsters often leave the level now.
     if (mon.pacified() && turns > random2(40) + 21)
     {
         make_mons_leave_level(&mon);
         return nullptr;
     }
 
-    // Following monsters don't get movement.
+    // Ignore monsters flagged to skip their next action
     if (mon.flags & MF_JUST_SUMMONED)
-        return &mon;
-
-
+        return nullptr;
```

Comment:
Are you sure about this? My plan was for `update_monster` to return the monster if it exists, regardless of whether it was updated by this function; returning nullptr only if the update causes the monster to not exist. If you make this change I think you'll need to also change at least one caller of `update_monster`.